File: BungieNetApi/Models/SingleComponentResponseOfDestinyPlugSetsComponent.cs
using System.Runtime.Serialization;

namespace GhostSharp.BungieNetApi.Models
{
    [DataContract]
    public class SingleComponentResponseOfDestinyPlugSetsComponent
    {
        /// <summary>
        /// Sockets may refer to a "Plug Set": a set of reusable plugs that may be shared across multiple sockets (or even, in theory, multiple sockets over multiple items).
        /// </summary>
        /// <summary>
        /// This is the set of those plugs that we came across in the users' inventory, along with the values for plugs in the set. Any given set in this component may be represented in Character and Profile-level, as some plugs may be Profile-level restricted, and some character-level restricted. (note that the ones that are even more specific will remain on the actual socket component itself, as they cannot be reused)
        /// </summary>
        [DataMember(Name = "data", EmitDefaultValue = false)]
        public DestinyPlugSetsComponent Data { get; set; }

        [DataMember(Name = "privacy", EmitDefaultValue = false)]
        public ComponentPrivacySetting Privacy { get; set; }

        /// <summary>
        /// If true, this component is disabled.
        /// </summary>
        [DataMember(Name = "disabled", EmitDefaultValue = false)]
        public bool Disabled { get; set; }


        public override bool Equals(object input)
        {
            return this.Equals(input as SingleComponentResponseOfDestinyPlugSetsComponent);
        }

        public bool Equals(SingleComponentResponseOfDestinyPlugSetsComponent input)
        {
            if (input == null) return false;

            return
                (
                    Data == input.Data ||
                    (Data != null && Data.Equals(input.Data))
                ) &&
                (
                    Privacy == input.Privacy ||
                    (Privacy != null && Privacy.Equals(input.Privacy))
                ) &&
                (
                    Disabled == input.Disabled ||
                    (Disabled != null && Disabled.Equals(input.Disabled))
                ) ;
        }
    }
}

File: pkg/models/SingleComponentResponseOfDestinyPlugSetsComponent.go
package bungieapigo

type SingleComponentResponseOfDestinyPlugSetsComponent struct {

	// Sockets may refer to a "Plug Set": a set of reusable plugs that may be shared across multiple
	// sockets (or even, in theory, multiple sockets over multiple items).
	// This is the set of those plugs that we came across in the users' inventory, along with the values
	// for plugs in the set. Any given set in this component may be represented in Character and
	// Profile-level, as some plugs may be Profile-level restricted, and some character-level
	// restricted. (note that the ones that are even more specific will remain on the actual socket
	// component itself, as they cannot be reused)
	Data DestinyPlugSetsComponent `json:"data"`

	Privacy ComponentPrivacySetting `json:"privacy"`

	// If true, this component is disabled.
	Disabled bool `json:"disabled"`
}
